frontend/src/concepts/pipelines/content/tables/experiments/ExperimentsTable.tsx

```diff
@@ -0,0 +1,119 @@
+import ExperimentsTableRow from '~/concepts/pipelines/content/tables/experiments/ExperimentsTableRow';
+import { SortableData } from '~/components/table/useTableColumnSort';
+import { getPipelineCoreResourceExperimentReference } from '~/concepts/pipelines/content/tables/utils';
+import useExperiments from '~/concepts/pipelines/apiHooks/useExperiements';
+import EmptyStateErrorMessage from '~/components/EmptyStateErrorMessage';
+
+type BaseTableProps<T> = Omit<TableProps<T>, 'data' | 'variant' | 'rowRenderer' | 'tbodyProps'>;
+
+type ExperimentsTableProps<T extends PipelineCoreResourceKF> = {
+  data: T[];
+  rowRenderer: (resource: PipelineCoreResourceKF) => React.ReactNode;
+} & BaseTableProps<T>;
+
+type Data<T> = {
+  experimentResource?: ExperimentKF;
+  name: string;
+  description: string;
+  containedResources: T[];
+};
+
+const DEFAULT_EXPERIMENT_NAME = 'Default';
+
+const ExperimentsTable = <T extends PipelineCoreResourceKF>({
+  data,
+  columns,
+  rowRenderer,
+  ...tableProps
+}: ExperimentsTableProps<T>) => {
+  const [experiments, loaded, loadError, refresh] = useExperiments();
+
+  if (loadError) {
+    return (
+      <EmptyStateErrorMessage title="Error displaying experiments" bodyText={loadError.message} />
+    );
+  }
+
+  if (!loaded) {
+    return <Spinner />;
+  }
+
+  const tableData: Data<T>[] = Object.values({
+    ...experiments.reduce<{ [experimentId: string]: Data<T> }>(
+      (acc, experiment) => ({
+        ...acc,
+        [experiment.name]: {
+          name: experiment.name,
+          description: experiment.description || '',
+          experimentResource: experiment,
+          containedResources: [],
+        },
+      }),
+      {},
+    ),
+    ...data.reduce<{ [experimentId: string]: Data<T> }>((acc, resource) => {
+      const experimentReference = getPipelineCoreResourceExperimentReference(resource);
+
+      let name: string;
+      let description = '';
+      let experimentResource: ExperimentKF | undefined;
+      if (experimentReference) {
+        experimentResource = experiments.find((e) => e.id === experimentReference.key.id);
+        if (!experimentResource) {
+          name = experimentReference.name || DEFAULT_EXPERIMENT_NAME;
+        } else {
+          name = experimentResource.name;
+          description = experimentResource.description || '';
+        }
+      } else {
+        name = DEFAULT_EXPERIMENT_NAME;
+        description = 'Runs that do not belong to an experiment';
+      }
+
+      let experimentData = acc[name];
+      if (!experimentData) {
+        experimentData = {
+          name,
+          description,
+          experimentResource: experimentResource,
+          containedResources: [],
+        };
+      }
+      experimentData.containedResources.push(resource);
+
+      return { ...acc, [name]: experimentData };
+    }, {}),
+  });
+  const tableColumns: SortableData<Data<T>>[] = columns.map((column) => ({
+    ...column,
+    sortable: false,
+  }));
```

Comment:
Now that I think of this -- I wonder if we can't do this statically with the columns instead of dynamically every render 🤔 Since we have the constants now.

Reply:
confused. isnt it still dynamic. we dont know T

Reply:
No worries, I see the issue you're setting. Let us leave it. I think TS is trying to protect us against a mistyped generic... but in reality the columns are only loosely related to data -- and in experiments table, the top-level data isn't even related to the columns.